adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/VertxBasedMqttProtocolAdapter.java

```diff
@@ -201,18 +201,16 @@ private void handleEndpointConnection(final MqttEndpoint endpoint) {
                     MqttUsernamePassword authObject = MqttUsernamePassword.create(endpoint,
```

Comment:
Yes, keeping the MqttEndpoint out of this class is better, I agree (was just a helper class as first step now).
No problem to change the name, but I might not fully understand your intentition : I wanted to keep exactly the username/password functionality (e.g. to parse the tenant from the username for multi-tenant setups) in one class and thus thought to give it the explicit name of it's responsibility would be a clear thing.
So, would you like it better to still keep it like that (without the endpoint in the signature) and only rename it to MqttAuth?
Or am I missing something?

A more generic class that is open for other MQTT authentication ways would be a followup step IMHO (first wanted to have the username/password in the code base).

WDYT?
Thanks!